flake.nix

```diff
@@ -7,8 +7,7 @@
 
     clickhouse-haskell.url = "github:nammayatri/clickhouse-haskell";
     clickhouse-haskell.inputs.common.follows = "common";
-    prometheus-haskell.url = "github:juspay/prometheus-haskell/more-proc-metrics";
-    prometheus-haskell.inputs.haskell-flake.follows = "common/haskell-flake";
+    prometheus-haskell.url = "github:vsaimanohar/prometheus-haskell/added-new-label-to-metrics";
```

Comment:
The `prometheus-haskell` input is not at all used in this flake..

shared-kernel does not use prometheus haskell _directly_, but it uses it _transitively_ via euler-hs -> euler-events-hs, so this is what you should change:

https://github.com/juspay/euler-events-hs/blob/214d436a83113c04b6f937c68488abf53beea4e8/flake.nix#L7-L8

But instead of modifying the transitive flake, you can do that by making use of the 'follows' feature: https://github.com/nammayatri/shared-kernel/pull/305